dlt/destinations/athena/athena.py

```diff
@@ -0,0 +1,253 @@
+        if args:
+            query, db_args = self._convert_to_old_pyformat(query, args)
+            if kwargs:
+                db_args.update(kwargs)
+        try:
+            cursor = self._conn.cursor(formatter=DLTAthenaFormatter())
+            cursor.execute(query, db_args)
+            yield cursor  # type: ignore
+        # TODO: figure out how to handle this
+        except DatabaseError as outer:
+            self.close_connection()
+            self.open_connection()
+            raise outer
+
+    def has_dataset(self) -> bool:
+        query = f"""SHOW DATABASES LIKE '{self.fully_qualified_dataset_name(escape=False)}';"""
+        rows = self.execute_sql(query)
+        return len(rows) > 0
+
+class AthenaClient(SqlJobClientBase):
+
+    capabilities: ClassVar[DestinationCapabilitiesContext] = capabilities()
+
+    def __init__(self, schema: Schema, config: AthenaClientConfiguration) -> None:
+        sql_client = AthenaSQLClient(
+            self.make_dataset_name(schema, config.dataset_name, config.default_schema_name),
+            config
+        )
+        super().__init__(schema, config, sql_client)
+
+
+    def _get_column_def_sql(self, c: TColumnSchema) -> str:
+        return f"`{c['name']}` {self._to_db_type(c['data_type'])}"
+
+    @classmethod
+    def _to_db_type(cls, sc_t: TDataType) -> str:
+        if sc_t == "wei":
+            return SCT_TO_HIVET["decimal"] % cls.capabilities.wei_precision
+        if sc_t == "decimal":
+            return SCT_TO_HIVET["decimal"] % cls.capabilities.decimal_precision
+        return SCT_TO_HIVET[sc_t]
+
+    @classmethod
+    def _from_db_type(cls, hive_t: str, precision: Optional[int], scale: Optional[int]) -> TDataType:
+        return HIVET_TO_SCT[hive_t]
+
+    def get_schema_by_hash(self, version_hash: str) -> StorageSchemaInfo:
+        query = f"""SELECT {self.VERSION_TABLE_SCHEMA_COLUMNS} FROM "{VERSION_TABLE_NAME}" WHERE version_hash = '{version_hash}';"""
+        try:
+            return self._row_to_schema_info(query)
+        except OperationalError:
+            return None
+
+    def _get_table_update_sql(self, table_name: str, new_columns: Sequence[TColumnSchema], generate_alter: bool) -> List[str]:
+        # TODO: get bucket var from outside
+        bucket = self.config.staging_config.bucket_url
+        dataset = self.sql_client.dataset_name
+        schema_name = self.schema.name
+        sql: List[str] = []
+
+        # for the system tables we need to create empty iceberg tables to be able to run, DELETE and UPDATE queries
+        iceberg = self.schema.tables[table_name]["write_disposition"] == "skip"
+        columns = ", ".join([self._get_column_def_sql(c) for c in new_columns])
+
+        # this will fail if the table prefix is not properly defined
+        table_prefix = path_utils.get_table_prefix(self.config.staging_config.layout, schema_name=schema_name, table_name=table_name)
+        location = f"{bucket}/{dataset}/{table_prefix}"
```

Comment:
this will come from `layout` right?

Reply:
it does, see function above that line

Reply:
Might be nice to trim trailing / from the bucket in case it's part of the string from the config?